src/meta/processors/job/AdminJobProcessor.cpp

```diff
@@ -27,45 +24,17 @@ void AdminJobProcessor::process(const cpp2::AdminJobReq& req) {
   }
   LOG(INFO) << __func__ << "() " << oss.str();
 
-  JobManager* jobMgr = JobManager::getInstance();
+  cpp2::AdminJobResult result;
+  auto errorCode = nebula::cpp2::ErrorCode::SUCCEEDED;
+  jobMgr_ = JobManager::getInstance();
+
   switch (req.get_op()) {
     case nebula::meta::cpp2::AdminJobOp::ADD: {
-      auto cmd = req.get_cmd();
-      auto paras = req.get_paras();
-      if (cmd == cpp2::AdminCmd::REBUILD_TAG_INDEX || cmd == cpp2::AdminCmd::REBUILD_EDGE_INDEX ||
-          cmd == cpp2::AdminCmd::STATS) {
-        if (paras.empty()) {
-          LOG(INFO) << "Parameter should be not empty";
-          errorCode = nebula::cpp2::ErrorCode::E_INVALID_PARM;
-          break;
-        }
-      }
-
-      JobID jId = 0;
-      auto jobExist = jobMgr->checkJobExist(cmd, paras, jId);
-      if (jobExist) {
-        LOG(INFO) << "Job has already exists: " << jId;
-        result.job_id_ref() = jId;
-        break;
-      }
-
-      folly::SharedMutex::WriteHolder holder(LockUtils::lock());
-      auto jobId = autoIncrementId();
-      // check if Job not exists
-      if (!nebula::ok(jobId)) {
-        errorCode = nebula::error(jobId);
-        break;
-      }
-
-      JobDescription jobDesc(nebula::value(jobId), cmd, paras);
-      errorCode = jobMgr->addJob(jobDesc, adminClient_);
-      if (errorCode == nebula::cpp2::ErrorCode::SUCCEEDED) {
-        result.job_id_ref() = nebula::value(jobId);
-      }
+      errorCode = addJobProcess(req, result);
```

Comment:
Will be 'addJobProcess' more clear and enough?

Reply:
My original intention is: for different job operations, separate out a separate interface.
So `addJobProcess` handles the operation of `add job`.
Of course, within the add job, there is its own check.